packages/client-sdk-nodejs/src/internal/pubsub-client.ts

```diff
@@ -238,16 +239,24 @@ export class PubsubClient extends AbstractPubsubClient<ServiceError> {
             options.subscription
           );
         }
-      } else if (resp?.heartbeat) {
+      } else if (resp.heartbeat) {
```

Comment:
Yeah, I inspected the type (`_SubscriptionItem`) when I saw that and I think `resp` will never be undefined; just that it may not contain any data if something goes wrong, and that would be handled by the `else` case.

```typescript
        constructor(data?: any[] | ({} & (({
            item?: _TopicItem;
            discontinuity?: never;
            heartbeat?: never;
        } | {
            item?: never;
            discontinuity?: _Discontinuity;
            heartbeat?: never;
        } | {
            item?: never;
            discontinuity?: never;
            heartbeat?: _Heartbeat;
        }))));
```